src/repr/src/scalar.rs

```diff
@@ -864,9 +864,13 @@ impl<'a> ScalarType {
     }
 }
 
-// TODO(benesch): the implementations of PartialEq and Hash for ScalarType can
-// be removed when decimal precision is either fixed or removed.
-
+/// NOTE: `ScalarType` equality is complicated by variants that store values.
+/// While we might want to handle stored values in one manner most of the time,
+/// that same approach might be insufficient in some cases.
+///
+/// When assessing unexpected behavior with type planning, it's worthwhile to
+/// see if the default equality is overridden in some other part of the code,
+/// e.g. when deciding how to handle potential casts between types.
```

Comment:
This seems good for now but long-term fishy to me. I think we want to eventually change `ScalarType` to have a normal derived `PartialEq` implementation, and then the things that want to be fuzzier about comparisons (e.g. ignoring the APD scale) can do _that_ with non-standard equality methods.

Reply:
Makes sense. Will be easier to shake out once the refactor lands.